homeassistant/components/foscam/camera.py

```diff
@@ -97,19 +196,56 @@ def enable_motion_detection(self):
         """Enable motion detection in camera."""
         try:
             ret = self._foscam_session.enable_motion_detection()
-            self._motion_status = ret == FOSCAM_COMM_ERROR
+
+            if ret != 0:
+                return
+
+            self._motion_status = True
         except TypeError:
             _LOGGER.debug("Communication problem")
-            self._motion_status = False
 
     def disable_motion_detection(self):
         """Disable motion detection."""
         try:
             ret = self._foscam_session.disable_motion_detection()
-            self._motion_status = ret == FOSCAM_COMM_ERROR
+
+            if ret != 0:
+                return
+
+            self._motion_status = False
         except TypeError:
             _LOGGER.debug("Communication problem")
-            self._motion_status = False
+
+    async def async_perform_ptz(self, movement, travel_time):
+        """Perform a PTZ action on the camera."""
+        _LOGGER.debug("PTZ action '%s' on %s", movement, self._name)
+
+        movement_function = {
+            DIR_UP: self._foscam_session.ptz_move_up,
```

Comment:
We could break out the dict to a module level constant if we write the method names as strings in the values of the dict. Then use `getattr` to get the method.

```py
MOVEMENT_ATTRS = {
    DIR_UP: "ptz_move_up",
    ...
}

...

movement_attr = MOVEMENT_ATTRS[movement]
movement_function = getattr(self._foscam session, movement_attr)
```

Reply:
I thought about doing this but didn't know if `getattr` was acceptable, I've seen some rejection previously, but yes this is cleaner so let's do it.